sdks/python/apache_beam/runners/portability/stager.py

```diff
@@ -656,25 +666,41 @@ def remove_dependency_from_requirements(
       for i in range(len(lines)):
         if not lines[i].startswith(dependency_to_remove):
           tf.write(lines[i])
+          requirements_to_install.append(lines[i].strip())
 
-    return tmp_requirements_filename
+    return tmp_requirements_filename, requirements_to_install
 
   @staticmethod
   @retry.with_exponential_backoff(
       num_retries=4, retry_filter=retry_on_non_zero_exit)
-  def _populate_requirements_cache(requirements_file, cache_dir):
-    # The 'pip download' command will not download again if it finds the
-    # tarball with the proper version already present.
-    # It will get the packages downloaded in the order they are presented in
-    # the requirements file and will download package dependencies.
-
-    # The apache-beam dependency  is excluded from requirements cache population
-    # because we  stage the SDK separately.
+  def _populate_requirements_cache(requirements_file, # type: str
+                                   cache_dir, # type: str
+                                   fetch_binary=False):
```

Comment:
To reduce code complexity, I think we can omit `fetch_binary` param from this flag. Currently, the only case when `fetch_binary=False` branch is used is custom container users who pass the requirements file. I think it's ok to change the behavior for these users as well, so that we download sources one-by-one from the requirements.txt file in this case, as opposed to download sources from the entire requirements file